main/src/library/Concurrent/Channel.flix

```diff
@@ -67,59 +72,50 @@ namespace Concurrent/Channel {
     /// conditions in `waitingPutters` comes from locks created by the get
     /// function or the select function.
     ///
-    pub enum Mpmc {
-        case Mpmc(
-            Int64, // id
-            Concurrent/ReentrantLock.ReentrantLock, // channelLock
-            Bool, // unBuffered
-            Int32, // maxQueueSize
-            MutDeque[Boxed, Static], // elementDeque
-            MutList[(Concurrent/ReentrantLock.ReentrantLock, Concurrent/Condition.Condition), Static], // waitingGetters
-            Concurrent/Condition.Condition // waitingPutters
-        )
-    }
+    pub enum Mpmc[a](
+        MpmcAdmin, // info, lock, etc
+        MutDeque[a, Static] // elementDeque
+    )
 
-    ///
-    /// Equality is based on the channel `id` and is needed for the `Order`
-    /// instance.
-    ///
-    instance Eq[Mpmc] {
-        pub def eq(x: Mpmc, y: Mpmc): Bool =
-        let Mpmc(a_id, _, _, _, _, _, _) = x;
-        let Mpmc(b_id, _, _, _, _, _, _) = y;
-        a_id == b_id
+    /// Returns the `MpmcAdmin` of `c`.
+    pub def mpmcAdmin(c: Mpmc[a]): MpmcAdmin = {
```

Comment:
You have this function, but don't use it consistently. Perhaps drop it?

Reply:
It was meant for lowering (`pub def`), It seems annoying to unpack the ADT in the compiler

Reply:
I also though about making the admin a record type since it could be nice to access fields directly

Reply:
I think we can afford to use some records, but that could be in another iteration.